emark/utils.py

```diff
@@ -116,6 +118,71 @@ def handle_data(self, data: str) -> None:
         if self.root:
             self.root.children.append(data)
 
+    @classmethod
+    def from_html(cls, html: str) -> HTML2TextParser:
+        """Parse HTML and return a HTML2TextParser instance.
+        >>> HTML2TextParser.from_html(html).text
+        """
+        parser = cls()
+        parser.feed(html)
+        return parser
+
+    @classmethod
+    def to_text(cls, html: str) -> str:
+        """Parse HTML and return plain text. Similar to Gmail.
+        >>> HTML2TextParser.to_text(html)
+        """
+        return str(cls.from_html(html))
+
+    @classmethod
+    def from_file(cls, path: str) -> HTML2TextParser:
+        """Parse HTML from a file and return a HTML2TextParser instance.
+        >>> HTML2TextParser.from_file(path).text
+        """
+        with open(path, "r") as file:
+            return cls.from_html(file.read())
+
+    @classmethod
+    def from_url(cls, url: str) -> HTML2TextParser:
+        """Parse HTML from a URL and return a HTML2TextParser instance. Requires requests.
+        >>> HTML2TextParser.from_url(url).text
+        """
+        import requests
+
+        return cls.from_html(requests.get(url).text)
```

Comment:
I'd use `urllib` here and not rely on requests as an extra dependency.
Furthermore, maybe add some exception handling. Network IO can fail, or the server does not respond HTML. I'd also recommend adding an HTTP [Accept](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept)-header.

Reply:
urllib seems fine and do the same